woocommerce/woocommerce
# [Experimental] Filter out filter options that would result in 0 products.

### Changes proposed in this Pull Request:

1. Whenever we display options to filter by, we don't show options that would result in a 0 product count, by filtering them out.

2. It made sense to migrate CollectionStockFilter to CheckboxList component as part of this.

Closes #43156



### How to test the changes in this Pull Request:

1. You'll need to have your env setup so that choosing one of the rating options would result in 0 products, same for stock and attribute filters
2. Add these filters to a page or template
3. See in the editor that we never render an option for these with a 0 count (this will only be visible in the checkbox list because dropdown in editor is non-interactive)
4. See in frontend we don't ever render options with 0 product count as well.



### Changelog entry




<details>

#### Significance




#### Type



-   [] Add - Adds functionality

#### Message 

[Experimental] In interactivity filters do not display filter options that would result in 0 products.

#### Comment 

</details>
